src/pydsl/scope.py

```diff
@@ -176,3 +209,13 @@ def new_scope(self, scope: Scope) -> Iterator[None]:
             yield
         finally:
             self.stack.pop()
+
+    @contextmanager
+    def new_binding_context(
+        self, binding_context: BindingContext
+    ) -> Iterator[None]:
+        try:
+            with self.stack[-1].new_binding_context(binding_context):
```

Comment:
this new_binding_context decorator reduces to a try-finally as well, so the outter try-finally isn't necessary